core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3285,13 +3287,31 @@ public void run(Reactor session) throws Exception {
      * Save the settings to a file.
      */
     public synchronized void save() throws IOException {
-        if(BulkChange.contains(this))   return;
-
-        if (initLevel == InitMilestone.COMPLETED) {
+        InitMilestone currentMilestone = initLevel;
+
+        if (!configLoaded) {
+            // someone is trying to save the config before all extensions are loaded (and possibly after as the task
+            // may run in parallel with other tasks.  OMG...!!! this is generally very bad and can lead to dataloss
+            LOGGER.log(Level.SEVERE,
+                       "An attempt to save Jenkins'' global configuration before it has been loaded has been "
+                       + "made during milestone " + currentMilestone
+                       + ".  This is indicative of a bug in the caller and may lead to full or partial loss of "
+                       + "configuration.",
+                       new IllegalStateException("call trace"));
```

Comment:
is there any need to create two `IllegalStateException`'s here? couldn't you just create one and then log and rethrow?

Reply:
I could, but I wanted the messages to be different.
the exception here is purely for a call stack in the log - and the longer message is not needed when we already have a detailed log statement